### PR TITLE
feat: make the password policy more configurable

### PR DESCRIPTION
**Pull request by RamiBerm**
Added two new password config fields: `min_password_length` and `identifier_similarity_check_enabled`.

## Related issue(s)
#970

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Review by aeneasr — CHANGES_REQUESTED** [review]
Sorry for the late review! This looks great! I think we should still enforce a lower limit for password length (e.g. 6).

Also could you please add a guide that explains how to use this feature in the docs/docs/guide directory? :)

**Comment by RamiBerm** [comment]
@aeneasr I've added a guide and reduced the default password length to 6

File: embedx/config.schema.json [hunk 1]
```diff
@@ -1128,6 +1128,18 @@
                       "description": "If set to false the password validation fails when the network or the Have I Been Pwnd API is down.",
                       "type": "boolean",
                       "default": true
+                    },
+                    "min_password_length": {
+                      "title": "Minimum Password Length",
+                      "description": "Defines the minimum length of the password.",
+                      "type": "integer",
+                      "default": 6
```
Comment (zepatrik) [inline embedx/config.schema.json:1136]: Please make the default `8`. We changed the default to be 8 in #2009, and it had quite some implications especially for automated tests and the login UI, so I would like to keep that default value.

File: driver/config/config.go [hunk 2]
```diff
@@ -998,10 +1002,12 @@ func (p *Config) ConfigVersion() string {
...
 	return &PasswordPolicy{
-		HaveIBeenPwnedHost:    p.p.StringF(ViperKeyPasswordHaveIBeenPwnedHost, "api.pwnedpasswords.com"),
-		HaveIBeenPwnedEnabled: p.p.BoolF(ViperKeyPasswordHaveIBeenPwnedEnabled, true),
-		MaxBreaches:           uint(p.p.Int(ViperKeyPasswordMaxBreaches)),
-		IgnoreNetworkErrors:   p.p.BoolF(ViperKeyIgnoreNetworkErrors, true),
+		HaveIBeenPwnedHost:               p.p.StringF(ViperKeyPasswordHaveIBeenPwnedHost, "api.pwnedpasswords.com"),
+		HaveIBeenPwnedEnabled:            p.p.BoolF(ViperKeyPasswordHaveIBeenPwnedEnabled, true),
+		MaxBreaches:                      uint(p.p.Int(ViperKeyPasswordMaxBreaches)),
+		IgnoreNetworkErrors:              p.p.BoolF(ViperKeyIgnoreNetworkErrors, true),
+		MinPasswordLength:                uint(p.p.IntF(ViperKeyPasswordMinLength, 6)),
```
Comment (zepatrik) [inline driver/config/config.go:1009]:
```suggestion
		MinPasswordLength:                uint(p.p.IntF(ViperKeyPasswordMinLength, 8)),
```

File: docs/docs/guides/password-policy.mdx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Kratos identities. They can be changed by modifying the following configuration
+parameters:
+
+```yaml title=path/to/kratos/config.yml
+selfservice:
+  methods:
+    enabled: true
+    config:
+      haveibeenpwned_enabled: true
+      min_password_length: 6
```
Comment (zepatrik) [inline docs/docs/guides/password-policy.mdx:16]:
```suggestion
      min_password_length: 8
```

File: docs/docs/guides/password-policy.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+#### `haveibeenpwned_enabled`
+
+If set to `true`, the password policy will check if the password has been found
+in the [Have I Been Pwned](https://haveibeenpwned.com/) database. The default
+value is `true`.
+
+#### `min_password_length`
+
+The minimum length of the password. The default value is `6`.
```
Comment (zepatrik) [inline docs/docs/guides/password-policy.mdx:28]:
```suggestion
The minimum length of the password. The default value is `8`.
```

**Review by zepatrik — CHANGES_REQUESTED** [review]

**Review by zepatrik — APPROVED** [review]
Nice, thank you :+1:

**Comment by kszafran** [comment]
Thanks from me, too ❤️. We were just discussing a need for this functionality internally.

**Comment by vinckr** [comment]
Hello @kszafran 
Congrats on merging your first PR in Ory 🎉 !
Your contribution will soon be helping secure millions of identities around the globe 🌏. 
As a small token of appreciation we send all our first time contributors a gift package to welcome them to the community. 
[Please drop me an email to claim your Ory swag!](mailto:vincent@ory.sh)

**Comment by kszafran** [comment]
@vinckr I'm just happy this got merged. The PR was authored by @RamiBerm. 😅

**Pull request merged**